test/AsioHelpersTest.cpp

```diff
@@ -4,269 +4,99 @@
-
-template <typename T, typename Rep, typename Period>
-net::awaitable<T> cancelAfter(net::awaitable<T> coroutine,
-                              std::chrono::duration<Rep, Period> duration) {
-  net::steady_timer timer{co_await net::this_coro::executor, duration};
-  co_await (std::move(coroutine) || timer.async_wait(net::use_awaitable));
-}
-// _____________________________________________________________________________
-
-// Checks that behavior is consistent for cancellation case
-TEST(AsioHelpers, resumeOnOriginalExecutorWhenCancelled) {
-  net::io_context ioContext{};
-  auto strand1 = net::make_strand(ioContext);
-  auto strand2 = net::make_strand(ioContext);
-  auto strand3 = net::make_strand(ioContext);
-  net::deadline_timer infiniteTimer{ioContext,
-                                    static_cast<net::deadline_timer::time_type>(
-                                        boost::posix_time::pos_infin)};
-
-  uint32_t sanityCounter = 0;
-
-  auto innerAwaitable = [&sanityCounter, strand1, strand2,
-                         &infiniteTimer]() -> net::awaitable<void> {
-    co_await net::post(net::bind_executor(strand2, net::use_awaitable));
-    // sanity check
-    EXPECT_FALSE(strand1.running_in_this_thread());
-    EXPECT_TRUE(strand2.running_in_this_thread());
-    sanityCounter++;
-    co_await infiniteTimer.async_wait(net::use_awaitable);
-  };
-
-  auto outerAwaitable = [&sanityCounter, &innerAwaitable, strand1, strand2,
-                         strand3]() -> net::awaitable<void> {
-    co_await net::post(net::bind_executor(strand1, net::use_awaitable));
-    // Sanity check
-    EXPECT_TRUE(strand1.running_in_this_thread());
-    EXPECT_FALSE(strand2.running_in_this_thread());
-    EXPECT_FALSE(strand3.running_in_this_thread());
-    EXPECT_THROW(co_await resumeOnOriginalExecutor(innerAwaitable()),
-                 boost::system::system_error);
-    // Verify we're on the strand the cancellation happened
-    EXPECT_FALSE(strand1.running_in_this_thread());
-    EXPECT_FALSE(strand2.running_in_this_thread());
-    EXPECT_TRUE(strand3.running_in_this_thread());
-    sanityCounter++;
-  };
-
-  net::co_spawn(strand3,
-                cancelAfter(outerAwaitable(), std::chrono::milliseconds(10)),
-                net::detached);
-
-  ioContext.run();
-
-  EXPECT_EQ(sanityCounter, 2);
-}
-
-// _____________________________________________________________________________
-
-// Checks that behavior is consistent for cancellation case
-TEST(AsioHelpers, resumeOnOriginalExecutorVoidOverloadWhenCancelled) {
-  net::io_context ioContext{};
-  auto strand1 = net::make_strand(ioContext);
-  auto strand2 = net::make_strand(ioContext);
-  auto strand3 = net::make_strand(ioContext);
-  net::deadline_timer infiniteTimer{ioContext,
-                                    static_cast<net::deadline_timer::time_type>(
-                                        boost::posix_time::pos_infin)};
-
-  uint32_t sanityCounter = 0;
```

Comment:
The sanity counter was there to ensure the assertions are even run in the first place. Of course you can just argue that other tests will fail if the function is not actually called, but this test will be green in that case because it did not run a single assertion